src/map/pc.c

```diff
@@ -5257,6 +5257,14 @@ static int pc_useitem(struct map_session_data *sd, int n)
 #endif
 			if( battle_config.item_restricted_consumption_type && sd->status.inventory[n].expire_time == 0 ) {
 				clif->useitemack(sd,n,sd->status.inventory[n].amount-1,true);
+
+				/// If Earth Spike Scroll is used while SC_EARTHSCROLL is active, there is a chance to don't consume the scroll. [Kenpachi]
+				if ((nameid == ITEMID_EARTH_SCROLL_1_3 || nameid == ITEMID_EARTH_SCROLL_1_5)
```

Comment:
I am not a fan of this, we already have so many hardcoded statement like this that we're trying to get rid of, adding more is not..fun, this should be converted to some item db flag.
also the message of the related commit is misleading 
>Implemented the consumption of 10 SP when using Earth Spike Scroll while SC_EARTHSCROLL is active.

while the change is for NOT consuming sp if i understand correctly.

Reply:
> I am not a fan of this, we already have so many hardcoded statement like this that we're trying to get rid of, adding more is not..fun, this should be converted to some item db flag.

I agree with you. But in my opinion this should be done in a separate PR for all/most of these hardcoded statements, because you can create a general solution then, instead of fixing just this one spot.

> also the message of the related commit is misleading
> 
> > Implemented the consumption of 10 SP when using Earth Spike Scroll while SC_EARTHSCROLL is active.
> 
> while the change is for NOT consuming sp if i understand correctly.

You probably mixed up some commits here. This part of the code is for not consuming the Earth Spike Scroll by chance and is not related to SP consumption.
 * Commit for not consuming the Earth Spike Scroll by chance while SC_EARTHSCROLL is active: https://github.com/HerculesWS/Hercules/pull/2616/commits/ff082a27bc3a6e6e86edbd8e66ff563244f6ba33
 * Commit for consuming 10 SP when using Earth Spike Scroll while SC_EARTHSCROLL is active: https://github.com/HerculesWS/Hercules/pull/2616/commits/803b1fae79d3a387791d9cb18c55cd580940c733